src/platform/silabs/wifi/icd/WifiSleepManager.cpp

```diff
@@ -92,32 +60,86 @@ CHIP_ERROR WifiSleepManager::RemoveHighPerformanceRequest()
 
 CHIP_ERROR WifiSleepManager::VerifyAndTransitionToLowPowerMode()
 {
-    VerifyOrReturnValue(mHighPerformanceRequestCounter == 0, CHIP_NO_ERROR,
-                        ChipLogDetail(DeviceLayer, "High Performance Requested - Device cannot go to a lower power mode."));
 
 #if SLI_SI917 // 917 SoC & NCP
+    if (mHighPerformanceRequestCounter > 0)
+    {
+        VerifyOrReturnError(ConfigureHighPerformance() == SL_STATUS_OK, CHIP_ERROR_INTERNAL);
+        return CHIP_NO_ERROR;
```

Comment:
Suggestion for all if with same logic
```suggestion
        return (ConfigureHighPerformance() == SL_STATUS_OK) ? CHIP_NO_ERROR : CHIP_ERROR_INTERNAL;
```